### PR TITLE
Changes related to creation of a modular backend in multiprecision [SyncWith: crypto3-multiprecision#70] 

### PR DESCRIPTION
**Pull request by martun**

### CODE REVIEW

File: .github/workflows/publish-results.yml [hunk 1]
```diff
@@ -10,4 +10,4 @@ on:
 jobs:
   call-reusable-workflow:
     name: Call Reusable Testing Callback Workflow
-    uses: NilFoundation/ci-cd/.github/workflows/reusable-crypto3-publish-result.yml@ca357baf14aa93d1ca6295b0397b2fc75be55c0d
+    uses: NilFoundation/ci-cd/.github/workflows/reusable-crypto3-publish-result.yml@v1.2.0
```
Comment (x-mass) [inline .github/workflows/publish-results.yml]: 1.2.1

File: .github/workflows/pull-request.yml
Comment (x-mass) [inline .github/workflows/pull-request.yml:1]: Some targets fail now?

File: test/integral.cpp [hunk 2]
```diff
@@ -49,15 +49,11 @@ struct unchecked_type {
...
-        nil::crypto3::multiprecision::cpp_int_backend<MinBits, MaxBits, SignType,
-                                                      nil::crypto3::multiprecision::unchecked, Allocator>,
+template<unsigned Bits, boost::multiprecision::expression_template_option ExpressionTemplates>
+struct unchecked_type<boost::multiprecision::number<
+    boost::multiprecision::cpp_int_modular_backend<Bits>, ExpressionTemplates>> {
+    typedef boost::multiprecision::number<
+        boost::multiprecision::cpp_int_modular_backend<Bits>,
         ExpressionTemplates>
         type;
 };
```
Comment (x-mass) [inline test/integral.cpp]: No need in this struct, every type is unchecked now

File: test/integral_fixed_size_container.cpp [hunk 3]
```diff
@@ -51,17 +51,9 @@ struct unchecked_type {
...
-    ExpressionTemplates>> {
-    typedef nil::crypto3::multiprecision::number<
-        nil::crypto3::multiprecision::cpp_int_backend<MinBits, MaxBits, SignType,
-                                                      nil::crypto3::multiprecision::unchecked, Allocator>,
-        ExpressionTemplates>
-        type;
+template<unsigned Bits, boost::multiprecision::expression_template_option ExpressionTemplates>
+struct unchecked_type<boost::multiprecision::number<boost::multiprecision::cpp_int_modular_backend<Bits>, ExpressionTemplates>> {
+    typedef boost::multiprecision::number<boost::multiprecision::cpp_int_modular_backend<Bits>, ExpressionTemplates> type;
 };
```
Comment (x-mass) [inline test/integral_fixed_size_container.cpp]: Ditto

**Review by x-mass — APPROVED** [review]

File: .github/workflows/publish-results.yml (hunk 1 above)
Comment (martun) [reply .github/workflows/publish-results.yml]: Changed.

File: .github/workflows/pull-request.yml
Comment (martun) [reply .github/workflows/pull-request.yml:1]: Locally all passes, hopefully will pass on CI soon.

File: test/integral.cpp (hunk 2 above)
Comment (martun) [reply test/integral.cpp]: You're right, removed.

File: test/integral_fixed_size_container.cpp (hunk 3 above)
Comment (martun) [reply test/integral_fixed_size_container.cpp]: Changed.

**Pull request merged**